transports/quic/src/transport.rs

```diff
@@ -472,7 +472,19 @@ impl<P: Provider> Listener<P> {
         }
 
         let endpoint_c = endpoint.clone();
-        let accept = async move { endpoint_c.accept().await }.boxed();
+        let accept = async move {
+            loop {
+                let incoming = match endpoint_c.accept().await {
+                    Some(incoming) => incoming,
+                    None => return None,
+                };
+                match incoming.accept() {
+                    Ok(connecting) => return Some(connecting),
+                    Err(_) => continue,
+                }
+            }
+        }
+        .boxed();
```

Comment:
this should still return the new `Incoming`, and the result of `incoming.accept` should be handled in [`Listener::poll_next`](https://github.com/libp2p/rust-libp2p/blob/master/transports/quic/src/transport.rs#L585), in case `Incoming::accept` returns an error `TransportEvent::ListenerError` should be returned